go/vt/vtadmin/cluster/config.go

```diff
@@ -284,6 +288,17 @@ func (cfg Config) Merge(override Config) Config {
 	return merged
 }
 
+func formatID(id string) string {
+	if strings.Contains(id, "_") {
```

Comment:
Are there any other characters that can occur in the input that will fail here as well? Is there any reference also to GRPC docs which describes what characters are valid and which ones are not?

Reply:
I couldn't find anything definitive in the gRPC repository, however:
- [gRPC Core](https://grpc.github.io/grpc/core/md_doc_naming.html) defines custom resolver names as following URI syntacx
- Further digging shows that gRPC naming followed ABNF, and specifically the `scheme` portion of the resolver does not seem to allow `_` underscores: https://datatracker.ietf.org/doc/html/rfc2396#section-3.1
    - we use the cluster ID as the scheme, which is why this PR replaces all the underscores in the cluster ID. For example `my_cluster_id://vtctld/` is a custom resolver name generated by vtadmin api

Reply:
I think I'd like to ship this fix, and then spend more time being thoughtful about:
- Testing the format stated in RFC 2396
- Implementing cluster ID validations following the format in the RFC, if it is indeed strict

Reply:
> cluster ID as the scheme

Which characters are allowed as a cluster ID? Should we add a validation on that so it wouldn't even be allowed to use anything that won't also fit into a scheme?

Reply:
https://datatracker.ietf.org/doc/html/rfc3986#appendix-A has the allowed characters as:

```
   scheme        = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
```

so we should just assert on that i think